### PR TITLE
[Kobaj] Add foundation work, graphql api calls, and disable auto logout.

### PR DESCRIPTION
**Pull request by kobaj**
Functionally this should be the same as what was previously in the repo, but with a bunch of upgrades, README, and foundation for next set of work.

Some of the code still needs refactored and tested, primarily
* popup.js
* disable_auto_logout.js
* event_page.js

But the following is 100% ready for consumption
* regfox/regfox_api.js 
* regfox/regfox_graphql_search_registrations_query.js 
* All the tests

### CODE REVIEW

File: public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "default_popup": "popup.html"
+  },
+  "permissions": [
+    "declarativeContent",
+    "scripting",
+    "storage"
+  ],
+  "host_permissions": [
+    "https://manage.webconnex.com/"
+  ],
```
Comment (cliffchapmanrbx) [inline public/manifest.json:16]: Do we want to add the regfox URLs in here as well just to future proof it? Manifest changes on things like that tend to invoke re-review when submitting extensions to the extension store.

**Review by Cellivar — APPROVED** [review]
Changes LGTM, a few observations:

* Including Lodash feels a little heavy-handed, though I admit I've never used it. Lately I've been shying away from using jQuery as well for general JS development and trying to stick to the APIs out of the box with modern web browsers. Ultimately this doesn't really matter that much so I'll not be blocking any reviews or anything on it.
* Should we add the regfox URLs to the manifest as well? Just for some future-proofing so we don't run into issues if we need to add it.
* The above said, I'd be interested in adding bootstrap for ease of slapping together the register UI when the time comes. It's no longer (as of 5.x) dependent on jQuery and apparently plays nice if it does happen to find jQuery present.
* Nice job adding tests! I'll try and add some when I get around to including more stuff.

**Comment by kobaj** [comment]
Lodash is admittedly big in its raw state, but the current file sizes aren't concerning to me.
```
      content_script.js ⏤  80.7 kB
 disable_auto_logout.js ⏤  1.38 kB
          event_page.js ⏤  98.6 kB
               popup.js ⏤  1.51 kB
              popup.css ⏤  884 B
             popup.html ⏤  482 B
              temp.html ⏤  370 B
```
If we hit ~mb I will implement treeshaking and other tricks to get the size reduced. 

---
I'm happy to add regfox urls. What are they?

---
I'll let you add bootstrap when you need it. I assume at that same time you can remove jQuery if you're able to. :)

---
Thank you! I love tests :D

**Comment by Cellivar** [comment]
> If we hit ~mb I will implement treeshaking and other tricks to get the size reduced.

Moreso leaning towards "does this really bring a lot to the table vs just using raw js?" side of 'heavy'. I'd say that jQuery has drifted towards "not really". Lodash seems to do more than I was expecting it to so I'll withdraw my comment on that in particular.

> I'm happy to add regfox urls. What are they?

I just went and looked and I think I'm mixing up memories, I thought there were RegFox-based URLs where we needed to manage something but I can't find them poking around the UI now. I suspect we're fine then.

> I assume at that same time you can remove jQuery if you're able to. :)

It very well might end up removed via attrition. The larger amount of new code coming in should make it easier to excise it at some point.

LGTM approval still stands, you're welcome to merge it in.

**Pull request merged**